File: Atcoder_#349/ruby/b_commencement.rb
def good_string?(s)
  # 文字の出現回数を数える
  char_count = Hash.new(0)
  s.each_char { |char| char_count[char] += 1 }

  # 出現回数ごとにその出現回数を持つ文字の種類数をカウントする
  count_freq = Hash.new(0)
  char_count.each_value { |count| count_freq[count] += 1 }

  # 出現回数ごとに条件をチェック
  count_freq.each_value do |freq|
    return false unless freq == 0 || freq == 2
  end

  true
end

# 標準入力から文字列を読み込む
s = gets.chomp

# 判定結果を出力する
puts good_string?(s) ? "Yes" : "No"

File: Atcoder_#349/javascript/b_commencement.js
const readline = require('readline');

// 標準入力を読み取るためのインターフェースを作成
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let inputLines = [];

// 標準入力を読み取るイベントリスナー
rl.on('line', (line) => {
  inputLines.push(line);
}).on('close', () => {
  // 入力データを処理する関数を呼び出し
  processInput(inputLines);
});

function processInput(input) {
  const s = input[0];
  console.log(goodString(s) ? "Yes" : "No");
}

function goodString(s) {
  // 文字の出現回数を数える
  const charCount = {};
  for (const char of s) {
    if (charCount[char]) {
      charCount[char]++;
    } else {
      charCount[char] = 1;
    }
  }

  // 出現回数ごとにその出現回数を持つ文字の種類数をカウントする
  const countFreq = {};
  for (const count of Object.values(charCount)) {
    if (countFreq[count]) {
      countFreq[count]++;
    } else {
      countFreq[count] = 1;
    }
  }

  // 出現回数ごとに条件をチェック
  for (const freq of Object.values(countFreq)) {
    if (freq !== 0 && freq !== 2) {
      return false;
    }
  }

  return true;
}
